### DISCUSSION TITLE
openssl api call to check legacy status

### DISCUSSION THREAD

**Discussion by timlegge** [openssl/openssl#24510] [Q&A] [answered]
Hi

I am wondering if there is a call that could be used to determine whether you can pass either of the following

            nid_long_name    "pbeWithSHA1And40BitRC2-CBC",
            nid_short_name   "PBE-SHA1-RC2-40,

            nid_long_name    "pbeWithSHA1And3-KeyTripleDES-CBC",
            nid_short_name   "PBE-SHA1-3DES"

and get a result that it is legacy or not? 

I can check https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html and see that RC4 is legacy but is there an api call that would confirm?

Tim

**Comment by t8m** [comment, marked as answer]
You can list algorithms in the legacy provider by using `openssl list -cipher-algorithms -propquery provider=legacy -provider legacy`

You can look how list_ciphers() is implemented in apps/list.c

**Comment by t8m** [comment]
However please note that for PBE algorithms you'll first need to identify which digest and cipher algorithms are they based upon.

**Reply by timlegge** [reply]
Thanks @t8m
